src/store/settings/settings.epic.ts

```diff
@@ -16,19 +16,19 @@ import { SettingsActions, SettingsSelectors } from './settings.reducers';
 
 const initEpic: AppEpic = (action$, state$) =>
   action$.pipe(
-    filter((action) => SettingsActions.initApp.match(action)),
+    filter(SettingsActions.initApp.match),
     tap(() => {
       const storageType = SettingsSelectors.selectStorageType(state$.value);
       DataService.init(storageType);
     }),
-    switchMap(() =>
+    switchMap(({ payload }) =>
       concat(
         of(ModelsActions.init()),
         of(AddonsActions.init()),
         of(ConversationsActions.init()),
         of(PromptsActions.init()),
         of(UIActions.init()),
-        of(FilesActions.init()),
+        payload.isLoggedin ? of(FilesActions.init()) : EMPTY,
```

Comment:
```suggestion
        payload.shouldLogin ? EMPTY : of(FilesActions.init()),
```
and we can apply the same for `ModelsActions.init()` and `AddonsActions.init()` to avoid additional requests

Reply:
Suggested changes applied.
ModelsActions.init() and AddonsActions.init() added to do request conditionaly